zds/forum/models.py

```diff
@@ -280,11 +296,55 @@ def last_read_post(self):
         except TopicRead.DoesNotExist:
             return self.first_post()
 
+    def resolve_last_read_post_absolute_url(self):
+        """resolve the url that leads to the last post the current user has read. If current user is \
+        anonymous, just lead to the thread start.
+
+        :return: the url
+        :rtype: str
+        """
+        user = get_current_user()
+        if not user.is_authenticated():
```

Comment:
tu veux dire user is None or not user.is_authenticated?

Le 14/07/2015 12:24, Pierre Beaujean a écrit :
>
> In zds/forum/models.py 
> <https://github.com/zestedesavoir/zds-site/pull/2913#discussion_r34554506>:
>
> > @@ -280,11 +296,55 @@ def last_read_post(self):
> >          except TopicRead.DoesNotExist:
> >              return self.first_post()
> >
> > +    def resolve_last_read_post_absolute_url(self):
> > +        """resolve the url that leads to the last post the current user has read. If current user is \
> > +        anonymous, just lead to the thread start.
> > +
> > +        :return: the url
> > +        :rtype: str
> > +        """
> > +        user = get_current_user()
> > +        if not user.is_authenticated():
>
> |if not user or user.is_authenticated()|`(j'en ai chié des barres dans 
> la ZEP-12 à cause de ça)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/zestedesavoir/zds-site/pull/2913/files#r34554506>.
>